drivers/include/lis2dh12.h

```diff
@@ -239,33 +244,71 @@ typedef struct {
 
 /**
  * @brief   Export the SAUL interface for this driver
+ * @{
  */
 extern const saul_driver_t lis2dh12_saul_driver;
+extern const saul_driver_t lis2dh12_saul_temp_driver;
+/** @} */
 
 #if MODULE_LIS2DH12_INT || DOXYGEN
 /**
- * @brief   Set the interrupt values in LIS2DH12 sensor device
+ * @brief   Configure a threshold event
+ *          An Interrupt will be generated if acceleration exceeds the set threshold
+ *          around the current reference value.
  *
- * @param[in] dev      device descriptor
- * @param[in] params   device interrupt configuration
- * @param[in] int_line number of interrupt line (LIS2DH12_INT1 or LIS2DH12_INT2)
+ * @param[in] dev       device descriptor
+ * @param[in] mg        acceleration in mg
+ * @param[in] us        time in µs for which the threshold must be exceeded
+ * @param[in] axis      bitmap of axis / events to be monitored
+ * @param[in] event     Event slot (1 or 2)
+ * @param[in] pin       Interrupt pin to use (LIS2DH12_INT1/LIS2DH12_INT2)
+ */
+void lis2dh12_cfg_threshold_event(const lis2dh12_t *dev,
+                                  uint32_t mg, uint32_t us,
+                                  uint8_t axis, uint8_t event, uint8_t pin);
+
+/**
+ * @brief   Configure a click event
+ *          A click event is generated when the acceleration exceeds the set threshold
+ *          for less than @p us_limit µs.
+ *          A double click event is generated if a second click event occurs within
+ *          @p us_window µs after the first one.
  *
- * @return  LIS2DH12_OK on success
- * @return  LIS2DH12_NOBUS on bus errors
+ * @param[in] dev       device descriptor
+ * @param[in] mg        acceleration in mg
+ * @param[in] us_limit  upper limit for click duration in µs
+ * @param[in] us_latency  dead time after click event in µs
+ * @param[in] us_window time after @p us_latency in which the second click event
+ *                      must occur to register as double click
+ * @param[in] click     bit map of click axis / types
+ * @param[in] pin       Interrupt pin to use (LIS2DH12_INT1/LIS2DH12_INT2)
  */
-int lis2dh12_set_int(const lis2dh12_t *dev, const lis2dh12_int_params_t *params, uint8_t int_line);
+void lis2dh12_cfg_click_event(const lis2dh12_t *dev, uint32_t mg,
```

Comment:
Is this type of event common application for this device? It seems somewhat specific.

Reply:
It's a hardware feature

![image](https://user-images.githubusercontent.com/1301112/116987292-2aba6b00-accf-11eb-8b5f-ec21ebeebeb1.png)